### PR TITLE
PMM-4879 Defaults file support.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

File: commands/management/add_mysql.go [hunk 1]
```diff
@@ -176,6 +180,35 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
...
+		path := cmd.DefaultsFile
+
+		type Config struct {
+			Client struct {
+				User     string
+				Password string
+			}
+		}
+
+		if strings.HasPrefix(path, "~/") {
```
Comment (BupycHuk) [inline commands/management/add_mysql.go]: Is it necessary step?

File: commands/management/add_mysql.go [hunk 2]
```diff
@@ -176,6 +180,35 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
 		}
 	}
 
+	if cmd.DefaultsFile != "" {
```
Comment (BupycHuk) [inline commands/management/add_mysql.go]: let's extract this block to separate method and write test for it.

File: commands/management/add_mysql.go [hunk 3]
```diff
@@ -176,6 +180,35 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
...
+			usr, err := user.Current()
+			if err != nil {
+				return nil, err
+			}
+			dir := usr.HomeDir
+			path = filepath.Join(dir, path[2:])
+		}
+
+		var cfg Config
+		err := gcfg.ReadFileInto(&cfg, path)
```
Comment (BupycHuk) [inline commands/management/add_mysql.go]: command shouldn't fail if file contains other configs. please read https://godoc.org/gopkg.in/gcfg.v1#hdr-Error_handling and https://godoc.org/gopkg.in/gcfg.v1#FatalOnly

**Review by BupycHuk — CHANGES_REQUESTED** [review]

**Comment by BupycHuk** [comment]
please don't forget to run dep ensure to include files related to new library.

File: commands/management/add_mysql.go (hunk 1 above)
Comment (JiriCtvrtka) [reply commands/management/add_mysql.go]: When I ran:
./bin/pmm-admin add mysql --defaults-file=~/my.sandbox.cnf MyDB 127.0.0.1:3306
without that, then I got:
open ~/my.sandbox.cnf: no such file or directory

Also I google little bit and found more people what do this. So I think yes, but I am Windows user, so I can be wrong.

**Comment by AlekSi** [comment]
We might want to use https://github.com/percona/go-mysql/tree/master/dsn there. @BupycHuk @JiriCtvrtka please check it out

File: commands/management/add_mysql.go (hunk 2 above)
Comment (JiriCtvrtka) [reply commands/management/add_mysql.go]: I wrote test which proof that username and password stay same if you dont specify defaults file, but I dont know if it is enough. Othervise I can write test, which find out working directory during test, create config file there, test it and delete it.

**Comment by JiriCtvrtka** [comment]
@AlekSi I didnt found support for cnf (ini) files, or I missed it?

**Pull request closed without merging**